cvmfs/ssl.h

```diff
@@ -0,0 +1,61 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_SSL_H_
+#define CVMFS_SSL_H_
+
+#include <string>
+#include <vector>
+
+#include "duplex_curl.h"
+#include "util/posix.h"
+
+#ifdef CVMFS_NAMESPACE_GUARD
+namespace CVMFS_NAMESPACE_GUARD {
+#endif
+
+inline unsigned int count_ssl_certificates(std::string directory) {
+  if (!DirectoryExists(directory)) return 0;
+
+  std::vector<std::string> dotpem = FindFilesBySuffix(directory, ".pem");
+  std::vector<std::string> dotcrt = FindFilesBySuffix(directory, ".crt");
+
+  return dotpem.size() + dotcrt.size();
+}
+
+inline bool AddSSLCertificates(CURL *handle) {
+  std::vector<std::string> cadirs;
+  const char *cadir = getenv("X509_CERT_DIR");
+  if (cadir != NULL) {
+    cadirs.push_back(std::string(cadir));
+  }
+  cadirs.push_back("/etc/grid-security/certificates");
+
+  // most systems store the certificates here
+  cadirs.push_back("/etc/ssl/certs/");
```

Comment:
We should not change the client default behavior.  We should have another bool parameter for this function that specifies if we look in system default directories.  We set this to false for the client and to true for the s3fanout.  We could check another environment variable in download.cc to optionally set this to true.

Reply:
I am not sure why we want to make this code more complex? The change in behavior is backward compatible and it helps the client to work on HTTPS servers.

Are we afraid of loosing performance? I believe we could measure the difference that this PR makes, but I believe it won't be that bad.

This will bring us to introduce another variable, if somebody will use cvmfs outside the grid they might prefer to use https and it means that the user will need to manage a more complex configuration.

Reply:
With the new code we are looking in more places for SSL certificates than we did previously.  This might eventually be what we want, but first it is a change in behavior for currently running clients.